qiskit_alice_bob_provider/plugins/state_preparation.py

```diff
@@ -49,16 +45,40 @@ def _reset_prep() -> Instruction:
     return Reset()
 
 
-def _get_unitary_synthesis(config: PassManagerConfig):
-    return CustomUnitarySynthesis(
-        config.basis_gates,
-        approximation_degree=config.approximation_degree,
-        coupling_map=config.coupling_map,
-        backend_props=config.backend_properties,
-        plugin_config=config.unitary_synthesis_plugin_config,
-        method=config.unitary_synthesis_method,
-        target=config.target,
-    )
+class CustomSolovayKitaev(TransformationPass):
+    def __init__(
+        self,
+        basic_approximations: List[GateSequence],
+        gates_to_decompose: FrozenSet[str],
+    ):
+        super().__init__()
+        self._basic_approximations = basic_approximations
+        self._gates_to_decompose = gates_to_decompose
+
+    def run(self, dag: DAGCircuit) -> DAGCircuit:
+        sk = SolovayKitaevDecomposition(self._basic_approximations)
+        for node in dag.named_nodes(*self._gates_to_decompose):
+            n_qubits = len(node.qargs)
+            min_qubits = 1
+            max_qubits = 1
+            if (n_qubits > max_qubits) or (n_qubits < min_qubits):
+                continue
+
+            unitary = node.op.to_matrix()
+            check_input = not isinstance(node.op, Gate)
+            approximate_dag = sk.run(
+                unitary, 3, return_dag=True, check_input=check_input
```

Comment:
I see here that we hardcode the recursion degree to 3 (vs qiskit where it's configurable), any particular reason?